### ISSUE TITLE
seems initial training python DeepSpeech.py not working 

### ISSUE THREAD

**Issue by kenyeung128** [mozilla/DeepSpeech#437]
Hi, after some minutes with python DeepSpeech.py, i got the following result and the program ends. Seems it doesn't work? Thanks.

    WER:    1.000000
    loss:   140.675095
    source: "she had your dark suit in greasy wash water all year"
    result: ""
--------------------------------------------------------------------------------
    WER:    1.000000
    loss:   140.675095
    source: "she had your dark suit in greasy wash water all year"
    result: ""
--------------------------------------------------------------------------------
    WER:    1.000000
    loss:   140.675095
    source: "she had your dark suit in greasy wash water all year"
    result: ""
--------------------------------------------------------------------------------
    WER:    1.000000
    loss:   140.675095
    source: "she had your dark suit in greasy wash water all year"
    result: ""
--------------------------------------------------------------------------------
    WER:    1.000000
    loss:   140.675095
    source: "she had your dark suit in greasy wash water all year"
    result: ""

**Comment by kdavis-mozilla** [comment]
What epoch did you train to?

**Comment by kdavis-mozilla** [comment]
Are you running form the latest master?

**Comment by kenyeung128** [comment]
yes, latest from Master, Epoch 0074

**Comment by kenyeung128** [comment]
FINISHED Epoch 0074   Overall epoch time: 0:00:11   Training time: 0:00:02

FINISHED Optimization   Overall time: 0:09:08   Training time: 0:03:41

**Comment by kdavis-mozilla** [comment]
- Is your master up to date? (Do a git update on the master branch)
- Have you changed parameters in `bin/run-ldc93s1.sh`? (It is by default set to run to epoch 50 not 74)
- Why are your epoch numbers not printing out? Have you changed other code in the repo?

**Comment by kdavis-mozilla** [comment]
For example a fresh checkout running master yields:
```
...
STARTING Epoch 0049
Training model...
Training loss=7.244015217 avg_cer=0.019230770 WER: 0.090909
--------------------------------------------------------------------------------
    WER:    0.090909
    loss:   7.244015
    source: "she had your dark suit in greasy wash water all year"
    result: "she had your dark suit n greasy wash water all year"
Hibernating training session into directory /Users/kdavis/.local/share/deepspeech
FINISHED Epoch 0049   Overall epoch time: 0:00:02   Training time: 0:00:01

FINISHED Optimization   Overall time: 0:01:34   Training time: 0:01:10

Testing model
Test loss=5.275577068 avg_cer=0.000000000 WER: 0.000000
Not publishing, missing some required environment variables: ['ds_website_server_fqdn', 'ds_website_server_root', 'ds_website_privkey', 'ds_website_username']
But maybe this is what you wanted, after all ...
```

**Comment by kenyeung128** [comment]
@kdavis-mozilla 
1. yes, i got the latest version.
2. I run 'python DeepSearch.py', not bin/run-ldc93s1.sh, but this could be the problem, because epoch is 75 and ds_n_hidden is 2048 in DeepSearch.py

so would you recommend not running 'python DeepSearch.py'? also how would it be able to support other languages? such as chinese/japanese, thanks.

**Comment by kenyeung128** [comment]
thanks @kdavis-mozilla a lot for your great work

**Comment by kdavis-mozilla** [comment]
@kenyeung128 For an initial test I'd suggest running `bin/run-ldc93s1.sh`
```bash
(tensorflow) kdaviss-MBP:DeepSpeech kdavis$ ./bin/run-ldc93s1.sh 
...
```
as it sets environments variables that work with the `ldc93s1` data set.

I'll make an issue #438 to update our `README.md` to suggest using `run-ldc93s1.sh`

**Comment by kdavis-mozilla** [comment]
@kenyeung128 Supporting other languages isn't too hard. Basically you have to create an importer for a corpus for the new data set and also make sure the softmax is outputting to the language's character set.

I know @reuben has already done a little Portuguese example. @reuben Does the example exist on github so @kenyeung128 could see how it's done?

**Comment by kdavis-mozilla** [comment]
@kenyeung128 I've made a pull request #439 changing the Read.me. It's just waiting on a review

**Comment by kenyeung128** [comment]
Thanks @kdavis-mozilla seems it's working now if I use ldc93s1.sh

But I still got the segmentation fault at the end, does it matter? I am using aws p2 for training. Not publishing, missing some required environment variables: ['ds_website_server_fqdn', 'ds_website_server_root', 'ds_website_privkey', 'ds_website_username']
But maybe this is what you wanted, after all ...
Segmentation fault (core dumped)

**Comment by kdavis-mozilla** [comment]
@kenyeung128 It shouldn't matter.

But, this is usually a signal of threads not being shutdown properly. I thought we'd gotten rid of all such problems. Let me file an issue.

What p2 instance are you using: p2.xlarge, p2.8xlarge, or p2.16xlarge?

**Comment by kenyeung128** [comment]
thanks a lot, it's p2.8xlarge

**Comment by reuben** [comment]
Yeah, my LapsBM importer is here: https://github.com/mozilla/DeepSpeech/blob/lapsbm/util/importers/lapsbm.py

There's not much in the way of language support there, it's just importing a Brazilian Portuguese corpus and training on it with no other changes. It still needs to support pt-BR specific characters, for example.

**Comment by kenyeung128** [comment]
thanks @kdavis-mozilla 

For this demo,
https://github.com/mozilla/DeepSpeech/tree/master/demos

I run bazel-bin/tensorflow_serving/model_servers/tensorflow_model_server --port=9000 --model_name=deepspeech --model_base_path=/path/to/deepspeech/export

what i should set for model_name and model_base_path after running run-ldc93s1.sh? I tried to set it to the model directory but it doesn't work. with errors

No versions of servable deepspeech found under base path /github/xdg_data_home/deepspeech

**Comment by kenyeung128** [comment]
@reuben thanks a lot, when you said it needs to support pt-BR specific characters, do you need to add extra code/dictionary for that?

Also, just wondering where and how to you use lapsbm.py? and if you could get some good performance with the model? Thank you.

**Comment by reuben** [comment]
Yes. When I wrote that importer I just wanted to do some quick tests, so I didn't bother with extended character support. You'd have to extend the encoding/decoding of the labels and increase the size of the output layer appropriately.

I did not obtain good results with LapsBM, the corpus is too small, so the network doesn't generalize well.

**Comment by kdavis-mozilla** [comment]
@kenyeung128 For this question

> For this demo,
> https://github.com/mozilla/DeepSpeech/tree/master/demos
> 
> I run bazel-bin/tensorflow_serving/model_servers/tensorflow_model_server --port=9000 --model_name=deepspeech --model_base_path=/path/to/deepspeech/export
> 
> what i should set for model_name and model_base_path after running run-ldc93s1.sh? I tried to set it to the model directory but it doesn't work. with errors
> 
> No versions of servable deepspeech found under base path /github/xdg_data_home/deepspeech

I'll defer to @Cwiiis who should be back from vacation tomorrow as he is most familiar with this code

**Comment by Cwiiis** [comment]
The model name will be 'deepspeech' unless you changed it in DeepSpeech.py, the directory will be the directory inside the export directory of the particular export - so, if I recall correctly, $ds_export_dir/00000001 or something along those lines.

**Comment by kdavis-mozilla** [comment]
@kenyeung128 Can you try the branch [issue440](https://github.com/mozilla/DeepSpeech/tree/issue440) to see if the segfault is removed there?

**Comment by kdavis-mozilla** [comment]
@kenyeung128 Did you get a chance to try the branch [issue440](https://github.com/mozilla/DeepSpeech/tree/issue440) to see if the segfault is removed there?

**Comment by kenyeung128** [comment]
looks like it's solved. thanks

**Comment by kenyeung128** [comment]
the training is converging
